ddtrace/internal/datastreams/kafka.py

```diff
@@ -24,7 +65,34 @@ def dsm_kafka_message_consume(instance, message):
     ctx = processor().decode_pathway(headers.get(PROPAGATION_KEY, None))
     ctx.set_checkpoint(["direction:in", "group:" + group, "topic:" + topic, "type:kafka"])
 
+    if instance._auto_commit:
+        # it's not exactly true, but if auto commit is enabled, we consider that a message is acknowledged
+        # when it's read.
+        reported_offset = message.offset() + 1 if isinstance(message.offset(), INT_TYPES) else -1
```

Comment:
this (and similar) changes fix an off-by-one error; when passing in a message to `commit()` confluent_kafka uses `message + 1` for the offset as the convention is to point the offset at the NEXT message.